src/modules/producer.js

```diff
@@ -0,0 +1,150 @@
+var utils = require('./utils'),
+  uuid = require('node-uuid'),
+  parsers = require('./message-parsers');
+
+var amqpRPCQueues = {};
+
+/**
+ * Create a RPC-ready queue
+ * @param  {string} queue the queue name in which we send a RPC request
+ * @return {Promise}       Resolves when answer response queue is ready to receive messages
+ */
+function createRpcQueue(queue) {
+  let rpcQueue = amqpRPCQueues[queue];
+  if (rpcQueue.queue) return Promise.resolve(queue);
+
+  //we create the callback queue using base queue name + appending config hostname and :res for clarity
+  //ie. if hostname is gateway-http and queue is service-oauth, response queue will be service-oauth:gateway-http:res
+  //it is important to have different hostname or no hostname on each module sending message or there will be conflicts
+  var resQueue = queue + ':' + this.conn.config.hostname + ':res';
+  rpcQueue.queue = this.conn.get().then((channel) => {
+    return channel.assertQueue(resQueue, { durable: true, exclusive: true })
+      .then((_queue) => {
+        rpcQueue.queue = _queue.queue;
+
+        //if channel is closed, we want to make sure we cleanup the queue so future calls will recreate it
+        this.conn.addListener('close', () => { delete rpcQueue.queue; createRpcQueue.call(this, queue); });
+
+        return channel.consume(_queue.queue, maybeAnswer.call(this, queue), { noAck: true });
+      })
+      .then(() => {
+        return queue;
+      });
+    })
+    .catch(() => {
+      delete rpcQueue.queue;
+      return utils.timeoutPromise(this.conn.config.timeout).then(() => {
+        return createRpcQueue.call(this, queue);
+      });
+    });
+
+  return rpcQueue.queue;
+}
+
+/**
+ * Get a function to execute on channel consumer incoming message is received
+ * @param  {string} queue name of the queue where messages are SENT
+ * @return {function}       function executed by an amqp.node channel consume callback method
+ */
+function maybeAnswer(queue) {
+  var rpcQueue = amqpRPCQueues[queue];
+
+  return (msg) => {
+    //check the correlation ID sent by the initial message using RPC
+    var corrIdA = msg.properties.correlationId;
```

Comment:
I don't know, thats an artifact of old code. I think corrId was stored globaly (:scream:) so there was a corrIdA... Will update that thanks